src/libpit/main.c

```diff
@@ -67,9 +67,9 @@ int pit_main(int argc, char *argv[], void (*callback)(int pe, void *data), void
   }
 
   sys_init();
-  debug_init(debugfile);
   ptr_init();
   thread_init();
+  debug_init(debugfile);
```

Comment:
This fixes a crash when the log level is set to 2, since the logging subsystem requires the ptr and thread subsystems to be initialized first.